wso2-synapse-unit-test-plugin/src/main/java/org/wso2/synapse/unittest/UnitTestCasesMojo.java

```diff
@@ -92,17 +130,22 @@ private void testCaseRunner() throws IOException {
      * @return list of files with their file paths
      */
     private ArrayList<String> getTestCasesFileNamesWithPaths(String testFileFolder) {
-
         ArrayList<String> fileNamesWithPaths = new ArrayList<>();
-        File folder = new File(testFileFolder);
-        File[] listOfFiles = folder.listFiles();
 
-        if (listOfFiles != null) {
-            for (File file : listOfFiles) {
-                String filename = file.getName();
-                if (filename.endsWith(".xml") || filename.endsWith(".XML")) {
+        if (testFileFolder.endsWith(Constants.XML_EXTENSION)) {
+            fileNamesWithPaths.add(testFileFolder);
+
+        } else if (testFileFolder.endsWith(Constants.TEST_FOLDER_EXTENSION)) {
+            String testFolderPath = testFileFolder.split("\\$")[0];
+            File folder = new File(testFolderPath);
+            File[] listOfFiles = folder.listFiles();
 
-                    fileNamesWithPaths.add(testFileFolder + File.separator + filename);
+            if (listOfFiles != null) {
```

Comment:
```folder.listFiles()``` will return empty array if no files in the directory. We do not need to do a null check here since testFolderPath must be path to a directory

Reply:
Done